assets/javascripts/discourse/models/docs.js

```diff
@@ -38,6 +39,13 @@ Docs.reopenClass({
     }
 
     return ajax(`/${docsPath}.json?${filters.join("&")}`).then((data) => {
+      const site = Site.current();
+      if (site.lazy_load_categories) {
```

Comment:
I'm abit concerned that we are exposing this site setting to plugins and am wondering if it is possible to just get plugins to sideload the data it needs and use it without any consideration for the "client side categories state".

Reply:
That makes sense. I applied your code suggestion.